asv_bench/benchmarks/benchmarks.py

```diff
@@ -1113,4 +1113,63 @@ def time_remove_categories(self, shape):
         execute(self.ts.cat.remove_categories(self.ts.cat.categories[::2]))
 
 
+class TimeSimpleReshape:
+
+    params = [get_benchmark_shapes("TimeSimpleReshape")]
+    param_names = ["shape"]
+
+    def setup(self, shape):
+        rows, cols = shape
+        k = 10
+        arrays = [
+            np.arange(rows // k).repeat(k),
+            np.roll(np.tile(np.arange(rows // k), k), 25),
+        ]
+        index = IMPL.MultiIndex.from_arrays(arrays)
+        self.df = IMPL.DataFrame(np.random.randn(rows, cols), index=index)
+        self.udf = self.df.unstack(1)
+        execute(self.df), execute(self.udf)
+
+    def time_stack(self, shape):
+        execute(self.udf.stack())
+
+    def time_unstack(self, shape):
+        execute(self.df.unstack(1))
+
+
+class TimeReplace:
+
+    params = [get_benchmark_shapes("TimeReplace")]
+    param_names = ["shape"]
+
+    def setup(self, shape):
+        rows, cols = shape
+        self.to_replace = {i: getattr(IMPL, "Timestamp")(i) for i in range(rows)}
+        self.df = IMPL.DataFrame()
+        for col in range(cols):
+            self.df["col" + str(col + 1)] = np.random.randint(rows, size=rows)
```

Comment:
```suggestion
        columns = [f"col{col + 1}" for col in range(cols)]
        self.df = IMPL.DataFrame(np.random.randint(rows, size=(rows, cols), columns=columns)
```

Reply:
Creating a dataframe in Modin by assigning columns in a loop results in inefficient internal partitioning, which in turn leads to poor performance.

Reply:
changed this to normal initialization of dataframe.